File: app/src/main/java/com/demo/room/bean/PersonAndClothes.java
package com.demo.room.bean;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

/**
 * Created by guoxiaodong on 2020/3/21 16:09
 */
public class PersonAndClothes {
    @Embedded
    public Person person;

    @Relation(parentColumn = "p_id", entityColumn = "father_id")
    public List<Clothes> clothesList;
}

File: app/src/main/java/com/gxd/demo/room/data/PersonAndClothes.kt
package com.gxd.demo.room.data

import androidx.room.Embedded
import androidx.room.Ignore
import androidx.room.Relation
import com.gxd.demo.room.table.Clothes
import com.gxd.demo.room.table.Person

data class PersonAndClothes
constructor(
    @Embedded
    var person: Person? = null,
    @Relation(parentColumn = "p_id", entityColumn = "owner_id")
    var clothesList: List<Clothes> = emptyList()
) {
    @Ignore
    constructor() : this(null)
}
